src/csync/csync_exclude.cpp

```diff
@@ -520,7 +548,62 @@ CSYNC_EXCLUDE_TYPE csync_excluded_traversal(CSYNC *ctx, const char *path, int fi
     return match;
 }
 
-CSYNC_EXCLUDE_TYPE csync_excluded_no_ctx(c_strlist_t *excludes, const char *path, int filetype) {
-  return _csync_excluded_common(excludes, path, filetype, true);
+CSYNC_EXCLUDE_TYPE ExcludedFiles::fullPatternMatch(const char *path, int filetype) const
+{
+    return _csync_excluded_common(_allExcludes, path, filetype, true);
+}
+
+auto ExcludedFiles::csyncTraversalMatchFun() const
+    -> std::function<CSYNC_EXCLUDE_TYPE(const char *path, int filetype)>
+{
+    return [this](const char *path, int filetype) { return this->traversalPatternMatch(path, filetype); };
+}
+
+/* Only for bnames (not paths) */
+static QString convertToBnameRegexpSyntax(QString exclude)
+{
+    QString s = QRegularExpression::escape(exclude).replace("\\*", ".*").replace("\\?", ".");
+    return s;
+}
+
+void ExcludedFiles::prepare()
+{
+    _nonRegexExcludes.clear();
+
+    // Start out with regexes that would match nothing
+    QString exclude_only = "a^";
```

Comment:
Maybe this can be done at the end   (`if exclude_only.isEmpty()) exclude_only = "a^";`)

Reply:
Can do.